packages/web/src/cookie-session.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ *
+ * Copyright 2024 Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { isIframe } from './utils'
+import { SessionState } from './types'
+
+export const COOKIE_NAME = '_splunk_rum_sid'
+
+const CookieSession = 4 * 60 * 60 * 1000 // 4 hours
+const InactivityTimeoutSeconds = 15 * 60
+
+export const cookieStore = {
+	set: (value: string): void => {
+		document.cookie = value
+	},
+	get: (): string => document.cookie,
+}
+
+export function parseCookieToSessionState(): SessionState | undefined {
+	const rawValue = findCookieValue(COOKIE_NAME)
+	if (!rawValue) {
+		return undefined
+	}
+
+	const decoded = decodeURIComponent(rawValue)
+	if (!decoded) {
+		return undefined
+	}
+
+	let sessionState: unknown = undefined
+	try {
+		sessionState = JSON.parse(decoded)
+	} catch {
+		return undefined
+	}
+
+	if (!isSessionState(sessionState)) {
+		return undefined
+	}
+
+	// id validity
+	if (
+		!sessionState.id ||
+		typeof sessionState.id !== 'string' ||
+		!sessionState.id.length ||
+		sessionState.id.length !== 32
```

Comment:
could you please use `SESSION_ID_LENGTH`? I mean can be done in the upcoming PR

Reply:
Sure, it will be improved in upcoming PR.